src/__tests__/test-utils-selectors.test.tsx

```diff
@@ -0,0 +1,141 @@
+              const filePath = selectorVars.get(expression.object.name)!;
+              if (expression.property.type === 'Identifier' && expression.property.name === 'default') {
+                onExtract(filePath, propertyName);
+              }
+            }
+          }
+          if (path.node.object.type === 'MemberExpression') {
+            if (path.node.property.type === 'Identifier') {
+              parseNested(path.node.object, path.node.property.name);
+            } else if (path.node.property.type === 'StringLiteral') {
+              parseNested(path.node.object, path.node.property.value);
+            } else if (path.node.property.type === 'TemplateLiteral') {
+              parseNested(path.node.object, buildTemplateString(path.node.property));
+            } else {
+              throw new Error('Unhandled selectors access type.');
+            }
+          }
+        },
+      },
+    } as PluginObj;
+  }
+  const source = fs.readFileSync(file, 'utf-8');
+  transformSync(source, { babelrc: false, configFile: false, plugins: [extractor] })?.code;
+}
+
+function extractComponentSelectors(file: string, usedProperties: string[], onExtract: (selector: string) => void) {
+  function extractor(): PluginObj {
+    return {
+      visitor: {
+        ObjectProperty(path: NodePath<types.ObjectProperty>) {
+          if (path.node.key.type === 'StringLiteral' && matchProperties(usedProperties, path.node.key.value)) {
+            if (path.node.value.type === 'StringLiteral') {
+              onExtract(trimSelectorHash(path.node.value.value));
+            } else {
+              throw new Error('Unexpected selector value format');
+            }
+          }
+        },
+      },
+    } as PluginObj;
+  }
+  const source = fs.readFileSync(file, 'utf-8');
+  transformSync(source, { babelrc: false, configFile: false, plugins: [extractor] })?.code;
+}
+
+function trimSelectorHash(selector: string) {
+  const splitSelector = selector.replace('.', '').split('_');
+  if (splitSelector.length >= 5) {
+    splitSelector.splice(splitSelector.length - 2, splitSelector.length);
+    return splitSelector.join('_');
+  }
+  return selector;
+}
+
+function getComponentNameFromFilePath(filePath: string) {
+  return filePath.match(/lib\/components\/([\w-]+)/)![1];
+}
+
+// Build string literal from template string replacing arguments with wildcards ("*").
+function buildTemplateString(node: types.TemplateLiteral) {
+  let literal = '';
+  for (const element of flatten(zip(node.quasis, node.expressions))) {
+    if (!element) {
+      continue;
+    } else if (element.type === 'TemplateElement') {
+      literal += element.value.raw;
+    } else if (element.type === 'Identifier') {
+      literal += '*';
+    } else if (element.type === 'NumericLiteral') {
+      literal += element.value;
+    } else {
+      throw new Error('Unhandled template literal structure.');
```

Comment:
Can we also add file name and line number to the error message to make it easier for other devs to investigate the issues?

Same applies to all errors here. Also, another reason to switch to the source parsing, to make it easier to track down issues.

Reply:
This should be visible in the stack trace. I also plan to move the extractor code to the test-utils package eventually so it can be reused across packages.

Reply:
> This should be visible in the stack trace

Can you show an example?

Reply:
Ah, no. Sorry, I misunderstood the suggestion. I will add file path and line number to every thrown exception.